hibernate6/pom.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Given we will support the new version through a new branch, why are we using the version number in our naming convention? Can we keep the file/directory naming consistent across hibernate 5.4 and hibernate 6.x branches? This will help when we cherry-pick commits and maintain both branches.

Some of the places where I am seeing this,
1. `spanner-hibernate6-dialect` directory name. Can we rename this to `spanner-hibernate-dialect` ?
2. `hibernate6` directory. Can we rename to `hibernate` ?
3. `spanner-hibernate6-samples` directory. Can we rename to `spanner-hibernate6-samples` ?